### PR TITLE
A version of the Newhall approximation that takes a nonconstant degree

### PR DESCRIPTION
**Pull request by pleroy**
… and returns the base class.  This is the one we'll use in `ContinuousTrajectory`.

### CODE REVIEW

File: numerics/newhall.hpp [hunk 1]
```diff
@@ -42,6 +43,17 @@ NewhallApproximationInMonomialBasis(std::vector<Vector> const& q,
                                     Instant const& t_max,
                                     Vector& error_estimate);
 
+// Same as above but the |degree| is not a constant expression.
+template<typename Vector,
+         template<typename, typename, int> class Evaluator>
+std::unique_ptr<Polynomial<Vector, Instant>>
```
Comment (eggrobin) [inline numerics/newhall.hpp]: `not_null`

File: numerics/newhall_body.hpp [hunk 2]
```diff
@@ -386,6 +386,127 @@ NewhallApproximationInMonomialBasis(std::vector<Vector> const& q,
...
+std::unique_ptr<Polynomial<Vector, Instant>>
+NewhallApproximationInMonomialBasis(int degree,
+                                    std::vector<Vector> const& q,
+                                    std::vector<Variation<Vector>> const& v,
+                                    Instant const& t_min,
+                                    Instant const& t_max,
+                                    Vector& error_estimate) {
+  switch (degree) {
+    case 3:
+      return std::make_unique<
```
Comment (eggrobin) [inline numerics/newhall_body.hpp]: `make_not_null_unique`

File: numerics/newhall.hpp (hunk 1 above)
Comment (pleroy) [reply numerics/newhall.hpp]: Done.

File: numerics/newhall_body.hpp (hunk 2 above)
Comment (pleroy) [reply numerics/newhall_body.hpp]: Done.

**Pull request merged**